patches/tModLoader/Terraria/ModLoader/ProjectileLoader.cs

```diff
@@ -295,18 +295,17 @@ private class HookList
 			}
 		}
 
-		private static HookList HookCanDamage = AddHook<Func<Projectile, bool>>(g => g.CanDamage);
+		private static HookList HookCanDamage = AddHook<Func<Projectile, bool?>>(g => g.CanDamage);
+
+		public static bool? CanDamage(Projectile projectile) {
 
-		public static bool CanDamage(Projectile projectile) {
-			if (projectile.modProjectile != null && !projectile.modProjectile.CanDamage()) {
-				return false;
-			}
 			foreach (GlobalProjectile g in HookCanDamage.arr) {
-				if (!g.Instance(projectile).CanDamage(projectile)) {
-					return false;
+				bool? canDamage = g.Instance(projectile).CanDamage(projectile);
+				if (canDamage.HasValue) {
+					return canDamage.Value;
 				}
 			}
-			return true;
+			return projectile.modProjectile?.CanDamage();
```

Comment:
As this uses nullable bools, this method should probably not short-circuit so simply. I think it should prefer returning false, if any hooks returned it? Then true, then null.